sql/rpl_info_table.cc

```diff
@@ -21,6 +21,46 @@
 #include "rpl_info_values.h"        // Rpl_info_values
```

Comment:
I'm not sure neither: this testcase would not test the memory leak bug exactly as you explain - it would just simply provide more coverage for the code paths executed and would serve as a correctness regresssion testcase.